### DISCUSSION TITLE
After using docker to start APISIX locally, the problem that it cannot communicate with the java plugin runner in the local Idea in debug mode

### DISCUSSION THREAD

**Discussion by sahara0506** [apache/apisix#5628] [Q&A]
环境：Mac OS，APISIX是使用Docker-compose启动的
现象：不使用PluginRunner时，可以正常访问接口，但是在使用PluginRunner之后，error.log中报以下错误信息：
2021/11/27 14:45:33 [error] 47#47: *54991 [lua] init.lua:617: phase_func(): failed to connect to the unix socket unix:/tmp/runner.sock: connection refused, client: 172.21.0.1, server: _, request: "POST /lotus2/provider HTTP/1.1", host: "192.168.101.8:9080"
2021/11/27 14:45:33 [warn] 47#47: *54991 [lua] plugin.lua:680: run_plugin(): ext-plugin-pre-req exits with http status code 503, client: 172.21.0.1, server: _, request: "POST /lotus2/provider HTTP/1.1", host: "192.168.101.8:9080"

请问是什么原因导致的？
/tmp/runner.sock，该文件已经映射到容器中，且已经授予777的权限，还是不行。

**Comment by tokers** [comment]
Could you provide the steps you launch the container and the stat info of the unix domain socket file.

**Reply by spacewander** [reply]
The process runs in host & the one runs in the docker are in the different net namespace. Even the socket file is shared via filesystem, they can't talk with each other.

**Reply by ms2008** [reply]
Docker for-mac does not support for sharing unix sockets, please use Linux instead.

ref: https://github.com/docker/for-mac/issues/483

**Comment by Bella-Liang** [comment]
请问如何能生成/tmp/runner.sock，我是centos7 docker环境
